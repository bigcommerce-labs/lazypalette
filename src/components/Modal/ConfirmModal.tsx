import React, { PureComponent } from 'react';

import ButtonInput from '../ButtonInput/ButtonInput';

import { Confirm, ConfirmBody, ConfirmButtons, ConfirmFooter } from './styles';
import Modal from './Modal';

interface ConfirmModalProps extends Partial<{
    body: string;
    title: string;
    primaryAction(): void;
    secondaryAction(): void;
    overlayClose(): void;
}> {}

const CTAButtons = ({ primaryAction, secondaryAction }: ConfirmModalProps) => (
    <ConfirmButtons>
        <ButtonInput
            onClick={primaryAction}
            classType="primary"
            type="button"
        >
            Cancel
        </ButtonInput>
        <ButtonInput
            onClick={secondaryAction}
            type="button"
        >
            OK
        </ButtonInput>
    </ConfirmButtons>
);

class ConfirmModal extends PureComponent<ConfirmModalProps> {
    static defaultProps: ConfirmModalProps = {
        title: 'Please Confirm:',
    };

    render() {
        const {
            body,
            primaryAction,
            secondaryAction,
            overlayClose,
            title,
        } = this.props;

        return (
            <Modal
                onClose={primaryAction}
                overlayClose={overlayClose}
                title={title}
                isTransparent={false}
            >
                <Confirm>
                    {body &&
                        <ConfirmBody>{body}</ConfirmBody>}
                    <ConfirmFooter>
                        <CTAButtons
                            primaryAction={primaryAction}
                            secondaryAction={secondaryAction}
                        />
                    </ConfirmFooter>
                </Confirm>
            </Modal>
        );
    }
}

export default ConfirmModal;
