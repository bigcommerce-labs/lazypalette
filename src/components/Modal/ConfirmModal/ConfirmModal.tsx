import React, { PureComponent } from 'react';

import ButtonInput from '../../ButtonInput/ButtonInput';

import { ConfirmButton, ConfirmButtons, ModalBody, ModalFooter, ModalView } from '../styles';
import Modal from '../Modal';

import { Messages } from '../constants';

interface ConfirmModalProps extends Partial<{
    title: string;
    primaryActionText?: string;
    primaryAction(): void;
    secondaryAction(): void;
    overlayClose(): void;
}> {}

const CTAButtons = ({ primaryAction, primaryActionText = Messages.Ok, secondaryAction }: ConfirmModalProps) => (
    <ConfirmButtons>
        <ConfirmButton>
            <ButtonInput
                onClick={secondaryAction}
                type="button"
                testId="cancel"
            >
                {Messages.Cancel}
            </ButtonInput>
        </ConfirmButton>
        <ConfirmButton>
            <ButtonInput
                classType="primary"
                onClick={primaryAction}
                type="button"
                testId={primaryActionText.replace(/\s+/g, '-').toLowerCase()}
            >
                {primaryActionText}
            </ButtonInput>
        </ConfirmButton>
    </ConfirmButtons>
);

class ConfirmModal extends PureComponent<ConfirmModalProps> {
    static defaultProps: ConfirmModalProps = {
        title: Messages.ConfirmTitle,
    };

    render() {
        const {
            children,
            primaryAction,
            secondaryAction,
            primaryActionText = Messages.Ok,
            overlayClose,
            title,
        } = this.props;

        return (
            <Modal
                onClose={secondaryAction}
                overlayClose={overlayClose}
                title={title}
                isTransparent={false}
            >
                <ModalView>
                    <ModalBody>{children}</ModalBody>
                    <ModalFooter>
                        <CTAButtons
                            primaryAction={primaryAction}
                            primaryActionText={primaryActionText}
                            secondaryAction={secondaryAction}
                        />
                    </ModalFooter>
                </ModalView>
            </Modal>
        );
    }
}

export default ConfirmModal;
