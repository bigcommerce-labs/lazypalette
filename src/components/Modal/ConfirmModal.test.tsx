import { shallow } from 'enzyme';
import React from 'react';

import ConfirmModal from './ConfirmModal';

describe('ConfirmModal', () => {
    it('renders', () => {
        const mockHandler = jest.fn();

        const mockBody = 'This theme will self-destruct in T-minus';

        const title = 'Please Confirm:';

        const confirmModal = shallow(
            <ConfirmModal
                body={mockBody}
                title={title}
                onClose={mockHandler}
                secondaryAction={mockHandler}
            />
        );

        expect(confirmModal).toMatchSnapshot();
    });
});
