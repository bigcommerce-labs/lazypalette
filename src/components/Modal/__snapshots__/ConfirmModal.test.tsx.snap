// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal renders 1`] = `
<Modal
  isTransparent={false}
  onClose={[MockFunction]}
  title="Please Confirm:"
>
  <styled.div>
    <styled.div>
      This theme will self-destruct in T-minus
    </styled.div>
    <styled.div>
      <styled.div>
        <ButtonInput
          classType="primary"
          onClick={[MockFunction]}
          type="button"
        >
          Cancel
        </ButtonInput>
        <ButtonInput
          onClick={[MockFunction]}
          type="button"
        >
          OK
        </ButtonInput>
      </styled.div>
    </styled.div>
  </styled.div>
</Modal>
`;
